### PR TITLE
raycasting support for THREE.CSS3DObject

### PR DESCRIPTION
**Pull request by jeromeetienne**
Hello,

This is an implementation of the THREE.Raycaster mechanism with THREE.CSS3DObject. It seems to work. i added a css3d_interactive_raycast.html. im still working on the examples itself

The reported intersection contains:
- intersection point
- distance to ray origin
- the intersected object
- the face/uv/faceIndex doesnt seem to make sense for css3d

What do you think ?

### CODE REVIEW

**Comment by WestLangley** [comment]
Sweet!

http://ci.threejs.org/api/pullrequests/8059/examples/index.html#css3d_interactive_raycast

File: examples/css3d_interactive_raycast.html [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+function onDocumentMouseMove( event ) {
+	event.preventDefault();
+	mouse.x = ( event.clientX / window.innerWidth ) * 2 - 1;
+	mouse.y = - ( event.clientY / window.innerHeight ) * 2 + 1;
+        
+        computeIntersect(mouse)
+}
+
+function computeIntersect(mouse){
+        var raycaster = new THREE.Raycaster();
```
Comment (WestLangley) [inline examples/css3d_interactive_raycast.html]: Create one Raycaster instance and reuse it.

**Comment by WestLangley** [comment]
For some reason, your `css3DObject` does not appear for me on OS X 10.11.3 Safari 9.0.5.

**Comment by mrdoob** [comment]
Hmm, we need a better example. This example could be done simpler with just adding a css :over to the css3d element...

**Comment by mrdoob** [comment]
Which, makes me wonder. What's the use case? One can use css :over for user interaction, but if it needs to be for physics, then one can put a `PlaneGeometry` in place...

**Comment by jeromeetienne** [comment]
@mrdoob About the usecase, we (at daqri doing augmented reality) needed that because we are using ray casting from camera position which are not rendered. we need that to handle click by reticle when rendering in stereo

## Details
- we have a world with 2 scene: one in css3d, one in webgl
- we are doing stereo rendering for both css3d and webgl to put that on a stereo display
- when clickling in stereo, the click position should happen on the mono camera (central one)
  - more info at https://www.nvidia.com/content/GTC-2010/pdfs/2010_GTC2010.pdf
- So, we got a left camera rendering left screen, right camera rendering right screen. but the click should happen on the central camera
- unfortunatly the central camera is never rendered. So using normal css interaction like :hover and all is not possible.

File: examples/css3d_interactive_raycast.html (hunk 1 above)
Comment (jeromeetienne) [inline examples/css3d_interactive_raycast.html]: thanks for reviewing it! i did the change you asked and remove some other debug stuff. i hope it is better now.

**Comment by mrdoob** [comment]
Why don't you have a normal scene/geometry to raycast to and apply the results to the css3d elements? I fear we are duplicating code needlessly.

**Comment by jeromeetienne** [comment]
@mrdoob if i understand you, so i would need to mirror the css3d scene into a scene with THREE.Mesh and planegeometry and do the raycasting there ?

- we actually considered this options. But to mirror a dynamic scene is not a simple task
- we need to mirror every css3d objects added/removed from the scene
- each objects position/rotation/scale should be mirrored too

all in all this make the process quite clumsy

**Comment by jeromeetienne** [comment]
@mrdoob the way i see it it is a matter of ubiquity: we use the same code for both type of scene.

Additionally the raycasting provide more detailed informations on the collision, especially the distance to the ray origin. We need this informations to compare and sort distance

**Comment by mrdoob** [comment]
Have you tried doing this?

```javascript
var geometry = new THREE.PlaneBufferGeometry( 100, 100 );
var mesh = new THREE.Mesh( geometry );
scene.add( mesh );

var css3d = new THREE.CSS3DObject();
mesh.add( css3d );

var renderer = new THREE.CSS3DRenderer();
renderer.render( scene );
```

With that setup, `Raycaster` will run on the normal geometries. You will animate normal meshes, but `CSS3DRenderer` will only render what it knows how to render, CSS3DObjects.

**Pull request closed without merging**